src/main/java/com/hillel/items_exchange/service/UserService.java

```diff
@@ -101,20 +101,23 @@ private UserDto mapUserToDto(User user) {
         return convertToDto(parent.getChildren(), ChildDto.class);
     }
 
-    public void addChildren(User parent, List<ChildDto> childrenDtoToAdd) {
+    public List<ChildDto> addChildren(User parent, List<ChildDto> childrenDtoToAdd) {
         final List<Child> childrenToSave = new ArrayList<>(convertAllTo(
                 childrenDtoToAdd, Child.class, ArrayList::new));
         addNewChildren(parent, childrenToSave);
         userRepository.save(parent);
+        return findDuplicatedChildAndConvertToDto(parent.getChildren(), childrenToSave);
```

Comment:
Naming the method in such way you're ruined SOLID principle of Single Responsibility. You have to split such method into two: find duplicates and convert dto. Or delegate mapping to another method and invoke it inside the first one.

Reply:
Okey, I will changed this